netfx_loader/DomainData.cs

```diff
@@ -1,54 +1,94 @@
 using System;
 using System.Collections.Generic;
 using System.Reflection;
+using System.Runtime.InteropServices;
 
 namespace ClrLoader
 {
     using static ClrLoader;
 
-    class DomainData : IDisposable
+    public static class DomainSetup
     {
         public delegate int EntryPoint(IntPtr buffer, int size);
+        public static void StoreFunctorFromDomainData()
+        {
+            var domain = AppDomain.CurrentDomain;
+            var assemblyPath = (string)domain.GetData("_assemblyPath");
+            var typeName = (string)domain.GetData("_typeName");
+            var function = (string)domain.GetData("_function");
+            var functor = GetFunctor(domain, assemblyPath, typeName, function);
+            domain.SetData("_thisFunctor", functor);
+        }
+
+        private static IntPtr GetFunctor(AppDomain domain, string assemblyPath, string typeName, string function)
+        {
+            var assemblyName = AssemblyName.GetAssemblyName(assemblyPath).Name;
+            var assembly = domain.Load(AssemblyName.GetAssemblyName(assemblyPath));
+            var type = assembly.GetType(typeName, throwOnError: true);
+            var deleg = Delegate.CreateDelegate(typeof(EntryPoint), type, function);
```

Comment:
You must ensure that this delegate stays alive, e.g. by attaching it to the `AppDomain` object via `SetData`.